plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -142,13 +144,58 @@ class CodeWhispererCodeScanManager(val project: Project) {
             return
         }
 
-        // Prepare for a code scan
-        beforeCodeScan()
+        //  If scanType is project
+        if (scanType == CodeWhispererConstants.SecurityScanType.PROJECT) {
+            // Prepare for a project code scan
+            beforeCodeScan(false)
 
-        // launch code scan coroutine
-        codeScanJob = launchCodeScanCoroutine()
+            // launch code scan coroutine
+            codeScanJob = launchCodeScanCoroutine(CodeWhispererConstants.SecurityScanType.PROJECT)
+        } else {
+            if (CodeWhispererExplorerActionManager.getInstance().isAutoEnabledForCodeScan()) {
+                // Prepare for a code scan
+                beforeCodeScan(true)
+                //  Add File Scan
+                codeScanJob = launchCodeScanCoroutine(CodeWhispererConstants.SecurityScanType.FILE)
+            }
+        }
+    }
+
+    private fun <T> debounce(
+        waitMs: Long = 300L,
+        coroutineScope: CoroutineScope,
+        destinationFunction: (T) -> Unit
+    ): (T) -> Unit {
+        var debounceJob: Job? = null
+        return { param: T ->
+            debounceJob?.cancel()
+            debounceJob = coroutineScope.launch {
+                delay(waitMs)
+                destinationFunction(param)
+            }
+        }
+    }
+
+    private fun createDebouncedCodeScan(
+        waitMs: Long = 300L,
+        coroutineScope: CoroutineScope
+    ): (String) -> Unit {
+        val debouncedCodeScan = debounce<String>(
+            waitMs = waitMs,
+            coroutineScope = coroutineScope,
+            destinationFunction = { scanType ->
+                runCodeScan(scanType)
+            }
+        )
+        return debouncedCodeScan
     }
```

Comment:
nit: do we need this? I think you can just directly call `debounce()` method on line 194 below